apps/st2-actions/controller.js

```diff
@@ -58,6 +58,17 @@ angular.module('main')
           })
           .groupBy('pack')
           .value();
+        _.forEach($scope.groups, function (value, key) {
+          $scope.groups[key] = {  
+            'list': value,
+            'icon': st2api.client.packFile.route(key+'/icon.png')
+          };
+          st2api.client.packFile.get(key+'/icon.png').catch(function () {
```

Comment:
Instead of making two requests for every pack (one for an actual icon and one to check if it's there) you can just use `client.packs.list()` from StackStorm/st2client.js#31 to get the list of all the packs and associated files